Kickstarter-iOS/Features/RootTabBar/RootTabBarViewController.swift

```diff
@@ -260,10 +260,11 @@ public final class RootTabBarViewController: UITabBarController, MessageBannerVi
       case let .search(index):
         _ = tabBarItem(atIndex: index) ?|> searchTabBarItemStyle
       case let .dashboard(index):
-        let featureFlaggedTabBarItemStyle = self
-          .isDashboardViewControllerDisplayable() ? dashboardTabBarItemStyle :
-          profileTabBarItemStyle(isLoggedIn: data.isLoggedIn, isMember: data.isMember)
-        _ = tabBarItem(atIndex: index) ?|> featureFlaggedTabBarItemStyle
+        let style = self
+          .isTabBarItemLastItem(for: index) ?
+          profileTabBarItemStyle(isLoggedIn: data.isLoggedIn, isMember: data.isMember) :
+          dashboardTabBarItemStyle
+        _ = tabBarItem(atIndex: index) ?|> style
```

Comment:
I had to do some rework here to configure the logic around getting the tab bar styling, which was still breaking.
Basically the logic is now - if the app is styling the dashboard tab bar item, it should not be the last tab bar item. If it is, then style it as profile.
So the cases are:
1) Logged in as backer (creator dashboard flag ON or OFF):
- The `setTabBarItemStyles` function never gets its' tab bar item set as `dashboard`, because the function that calls this one checks if the user is a `member` first.
2) Logged in as a creator (creator dashboard flag ON):
- If set in Beta tools as ON then go from backer --> creator, then the dashboard is visible.
- Turning this flag OFF in beta tools doesn't reflect the hiding of the dashboard until the user relaunches the app.
3) Logged in as a creator (creator dashboard flag OFF):
- If set in Beta tools as ON then go from backer --> creator, then the dashboard is not visible.
- Turning this flag ON in beta tools doesn't reflect the showing of the dashboard until the user relaunches the app.

Validate my logic here. I tested it, but if you can find a missing step I'm happy to rework this.

Reply:
Interesting that I didn't see this bug, but I've tested the current logic for all cases listed and the app is working as expected. Good to go 👍